### PR TITLE
plexus-cipher 3.0.0

### PR DESCRIPTION
**Pull request by cstamas**
Cipher is a small simple (nearly trivial) component that is able to encrypt/decrypt strings using provided "master password".

This commit moves it to version 3.0.0 denoting a major (and backward incompatible jump.

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

File: src/test/java/org/codehaus/plexus/components/cipher/internal/DefaultPlexusCipherTest.java [hunk 1]
```diff
@@ -88,6 +91,7 @@ void testDefaultAlgorithmExists() throws Exception {
 
     // -------------------------------------------------------------
 
+    @Disabled("This test is not really a test")
```
Comment (gnodet) [inline src/test/java/org/codehaus/plexus/components/cipher/internal/DefaultPlexusCipherTest.java:94]: Maybe delete it then ?

File: src/test/java/org/codehaus/plexus/components/cipher/internal/PBECipherTest.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ class PBECipherTest {
 
     final String clearText = "veryOpenText";
 
-    final String encryptedText = "F7eMV2QRQF4H0ODCA1nrTGUWacCXVvPemSjaQjGbO6U=";
+    final String encryptedText = "ce/l2ofOiSELRT1WAjOyNoZbG+2FQcrlOKEdDr5mi6esyR2LfvBY855yxW5bqHZi";
```
Comment (gnodet) [inline src/test/java/org/codehaus/plexus/components/cipher/internal/PBECipherTest.java]: Why has the encrypted value changed ?

**Review by gnodet — COMMENTED** [review]
LGTM.  If I summarise:
 * change package name to org.codehaus.plexus (which sounds weird given Codehaus does not exist anymore...)
  * change the default cipher from `AES/CBC/PKCS5Padding` to `AES/GCM/NoPadding`, why ?

**Comment by cstamas** [comment]
> LGTM. If I summarise:
> 
> * change package name to org.codehaus.plexus (which sounds weird given Codehaus does not exist anymore...)
> * change the default cipher from `AES/CBC/PKCS5Padding` to `AES/GCM/NoPadding`, why ?

The package name change is IMO okay, many (today GH, former Codehaus) orgs still publish under org.codehaus package, and this has been well established (like older groovy versions that are still maintained). I don't find this problematic. This GH org "owns" the org.codehaus.plexus G on Central, so using same package is okay as well IMO. OTOH, I find it more problematic to use package name of existing private entity/company.

The cipher changes have been undone, I was just reading and trying out some last night. Now there is only one change, they secret key alg is changed to `PBKDF2WithHmacSHA512` as recommended since Java 8.

The encrypted value changes really does not matter, as 2.1.0 release _already broke backward compat_, see https://github.com/codehaus-plexus/plexus-cipher/pull/23 If we'd move from 2.0 to 2.1.0, users would need to _reencrypt everything they have encrypted anyway_. So IMO this change is ok as well.

Finally, to signify all that above, but also the fact of moving to Java 17, am moving version to 3.0.0.

**Comment by cstamas** [comment]
In fact, just checked: _nothing mandates for Java 17_, so I undid that bit as well. Having this lib 8+ is just extra benefit.

**Comment by slachiewicz** [comment]
To me GCM/No padding also looks good. 
We can assume that version 2.1.0 was never used in our components

**Comment by slachiewicz** [comment]
If not under codehaus, what other alternatives?
Move it to any other foundation or an organisation? Together with all other components like Modello?

**Review by gnodet — APPROVED** [review]

**Pull request merged**